src/server/game/Entities/Player/Player.cpp

```diff
@@ -11660,7 +11660,12 @@ Item* Player::StoreNewItem(ItemPosCountVec const& pos, uint32 itemId, bool updat
         for (int32 bonusListID : bonusListIDs)
             item->AddBonuses(bonusListID);
 
-        item = StoreItem(pos, item, update);
+        // Since 4.x bags are getting equipped automaticly when they have no bind on equip boundary
+        // This includes looted bags, bags from quest rewards and purchased bags from vendors
+        if (item->IsBag() && item->GetTemplate()->GetBonding() != BIND_ON_EQUIP)
```

Comment:
wouldn't it be better to move IsBag() and GetBonding() checks inside Player::StoreItem() since it is called in both cases anyway?

Reply:
You can use ```pItem->HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_NEW_ITEM);``` in Player::StoreItem(), no need to have an additional argument lootedBag.

Reply:
not supported on 335 or 4.x so bad idea.